oauth2_http/java/com/google/auth/oauth2/GoogleCredentials.java

```diff
@@ -54,6 +54,7 @@
   static final String QUOTA_PROJECT_ID_HEADER_KEY = "x-goog-user-project";
   static final String USER_FILE_TYPE = "authorized_user";
   static final String SERVICE_ACCOUNT_FILE_TYPE = "service_account";
+  static final String IMPERSONATED_SERVICE_ACCOUNT = "impersonated_service_account";
```

Comment:
This constant does improve on the literal string. I suggest just using the string.

Reply:
I am debating on these two options. I think even though the constant is not more readable than the string itself. I like the way we define all credential types in one place. So people can see all supported types there. I am not very familiar with Java convention, I trust your judgement on this. Please let me know if you still want me to make this change.

Reply:
That's something that belongs in documentation. Don't rely on the code for that.

Reply:
Done